0000-interledger-protocol-4/0000-interledger-protocol-4.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Interledger Protocol V4 (ILPv4)
+draft: 1
+---
+
+# Interledger Protocol V4
+
+Interledger is a protocol for sending packets of money across different payment networks or ledgers. ILPv4 is a simplification of previous versions of the protocol designed primarily for "penny switching", or routing large volumes of smaller-value packets. ILPv4 can be integrated with any type of ledger, including those not built for interoperability, and it can be used with a variety of higher-level protocols that implement features ranging from quoting to sending larger amounts of value with chunked payments. 
```

Comment:
Maybe say `using "penny switching" to limit risk instead of expensive atomic swaps`

Reply:
There are a bunch of reasons we opted for penny switching, risk being one of them. To me, that phrasing seems to bring up more questions than it answers (for example, what makes atomic swaps expensive?) and I'm not sure this document is the place to speak to all of those points

Reply:
It is emergent but we have made a number of assumptions that are based on relatively small packet amounts:

- Connectors can severely limit their customers' bandwidths to limit their risk, without making their customers' payments fail (because the customers will work around that)
- Sending packets is free or of negligible cost, so we can use them liberally for things like quoting
- We don't need a way to express how exchange rates vary based on the payment amount and/or hold time
- As someone sending packets through the open Interledger, you won't be able to rely on the assumption that your payment can go through in one chunk and you might be told it's too big

And some of the implementation assumptions:
- Ledger protocols are now mostly request/response, meaning the response (fulfillment) isn't retried if it doesn't go through the first time
- Connectors don't store prepared packets in a db, because the assumption is that if your system dies you just lose all the packets that are in-flight and the timeout is too short for you to bring the system up and submit the fulfillments anyway
- Connectors have relatively low maximum hold times by default and as a connector you should be prepared to forward packets that have less than 30 seconds before they expire (you won't fare very well if you need to build in a long expiry window because your ledger is slower than that)